site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+    <p><strong>Selections that demand the users consent, like approving terms and conditions <em>must</em> never be pre-selected as default.</strong></p>
+</div>
+
+---
+
+## Providing instructions
+To help the user enter the **right information in the right format**, guidance can be provided on different levels:
+
+1. **Form main and sub-heading** summarise the purpose of the form
+2. **Form wide instructions** to help the user to better understand the functionality of the form
+3. **Contextual instructions** for helping the user to fill specific form controls
+
+---
+
+### 1. Form heading and sub-heading
+The form heading **summarises the purpose of the form** and indicates the user where the form begins. 
+
+- The *main heading* should clearly and concisely **describe the purpose of the form**. 
+    - Full page forms should always have a main heading, but from smaller inline forms embedded to page content, the heading can be omitted.
+- If the purpose of the form cannot be sufficiently summarised by the main heading, it can **elaborated with a *sub-heading***.
+
+<Image src="../../static/patterns/form/form-guiding-headings@2x.png" alt="Form headings" style="max-width:432px;" viewable /> 
+
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li><strong>If the form has a main heading, it <em>must</em> be the first element</strong>
+            <ul style="margin-top:0;">
+                <li>The main heading <em>should</em> preferably be h1 element, even if visually styled as a smaller level heading.</li>
+            </ul>
+        </li>
+        <li>The the sub-heading <em>should</em> placed next after the main heading, and preferably be a h2 element.</li>
+        <li><strong>Keep it short and descriptive!</strong>
+            <ul style="margin-top:0;">
+                <li>Main heading <em>should</em> be less than five words.</li>
+                <li>Sub-heading <em>should</em> be less than two to three sentences.</li>
+            </ul>
+        </li>
+    </ul>
+</div>
+
+---
+
+### 2. Form wide instructions
+To help the user better understand the functionality of the form, instructions and other **information about filling the form and required information** should be provided in the beginning of the form. 
+
+<Image src="../../static/patterns/form/form-guiding-instructions@2x.png" alt="Form wide instructions" style="max-width:432px;" viewable /> 
+
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
```

Comment:
> Do

Näihin liittyy myös se, että ohjeistuksen mukaan näitä ei saisi olla sivulla montaa samaan aikaan. Eli periaatteessa ainoa sallittu paikka näille on formin (tai formisivun) alussa. Vai onko tämä liian rajoittava ohje?

Reply:
Hmm. Vähänhän se rajoittaa. Esimerkiksi joissain pitkissä formeissa voisi periaatteessa olla järkevää antaa ykstiyiskohtaisempia osiokohtaisia ohjeita heti väliotsikon perässä… Mutta jos tähän ohjeistetaan käytettäväksi notification-komponenttia, niin silloin niitä pitäisi tietty olla vain yksi.